### PR TITLE
Adding chunking settings to GoogleVertexAiService, AzureAiStudioService, and AlibabaCloudSearchService

### PR DESCRIPTION
**Pull request by dan-rubinstein**
## Description
This change introduces:
- Chunking settings as a new setting to provide when creating embedding inference endpoints in CohereService, AmazonBedrockService, and AzureOpenAIService.

These features are currently hidden behind a feature flag for 8.16. 

Note: Chunking currently only runs when large documents are ingested to an index with an inference field.

## Testing
- Ran unit tests and ensured that all succeed
- Tested the following cases with each service with the feature flag enabled:
  - Creating an embedding endpoint with no `chunking_settings` provided generates a model with default chunking settings (keeps existing word based chunking strategy with default configurations).
  - Creating an embedding endpoint with `chunking_settings` for a word based chunking strategy provided generates a model that uses the word based chunking strategy.
  - Creating an embedding endpoint with `chunking_settings` for a sentence based chunking strategy provided generates a model that uses the sentence based chunking strategy.
  - Creating a non-embedding endpoint (when applicable) works.
- Tested the following cases with the feature flag disabled:
  - Creating an embedding endpoint with no `chunking_settings` provided generates a model with null chunking settings and chunking still runs successfully with the previously default word based chunking strategy.
  - Attempting to create an embedding endpoint with `chunking_settings` provided throws an exception.
  - Attempting to create a non-embedding endpoint works.

### CODE REVIEW

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/alibabacloudsearch/AlibabaCloudSearchService.java [hunk 1]
```diff
@@ -238,11 +266,22 @@ protected void doChunkedInfer(
...
-            inputs.getInputs(),
-            EMBEDDING_MAX_BATCH_SIZE,
-            EmbeddingRequestChunker.EmbeddingType.FLOAT
-        ).batchRequestsWithListeners(listener);
+        List<EmbeddingRequestChunker.BatchRequestAndListener> batchedRequests;
+        if (ChunkingSettingsFeatureFlag.isEnabled()) {
+            batchedRequests = new EmbeddingRequestChunker(
+                inputs.getInputs(),
+                EMBEDDING_MAX_BATCH_SIZE,
+                EmbeddingRequestChunker.EmbeddingType.FLOAT,
```
Comment (davidkyle) [inline x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/alibabacloudsearch/AlibabaCloudSearchService.java:274]: Alibaba supports sparse embeddings. If the task type is sparse then the embedding type should also be sparse

**Review by davidkyle — COMMENTED** [review]
Looks good, I see an issue in the existing code that would not chunk for sparse embeddings properly, please can you look at that

**Comment by dan-rubinstein** [comment]
@elasticmachine merge upstream

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/alibabacloudsearch/AlibabaCloudSearchService.java (hunk 1 above)
Comment (dan-rubinstein) [reply x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/alibabacloudsearch/AlibabaCloudSearchService.java]: Good catch, I've updated this and also noticed that we were missing some overrides in the text embedding service settings for Alibaba which were preventing us from using it for semantic text fields. I've updated these both now.

**Review by davidkyle — APPROVED** [review]
LGTM

**Comment by dan-rubinstein** [comment]
@elasticmachine merge upstream

**Comment by dan-rubinstein** [comment]
@elasticmachine merge upstream

**Comment by dan-rubinstein** [comment]
@elasticmachine merge upstream

**Pull request merged**